File: 01_make-aws-lambda-handler-testable/nodejs/code_initial/src/main.js
const AWS = require('aws-sdk')

const cost_center = process.env.COST_CENTER;
const client = new AWS.Lambda();

exports.handler = async function (event, context) {
    if (!cost_center) {
        throw new Error("expecting cost center ID")
    }

    if (!event.arn) {
        throw new Error("expecting 'arn' key")
    }

    await client.tagResource({
        Resource: event.arn,
        Tags: {
            CostCenter: cost_center
        }
    }).promise()

    response = await client.listTags({
        Resource: event.arn
    }).promise()

    return response.Tags
}

File: 01_make-aws-lambda-handler-testable/python/code_initial/src/main.py
import boto3
import os

cost_center = os.environ["COST_CENTER"]
client = boto3.client("lambda")


def handler(event: dict, context: object) -> dict:
    assert cost_center, "expecting cost center ID"

    client.tag_resource(
        Resource=event["arn"], Tags={"CostCenter": cost_center},
    )

    response = client.list_tags(Resource=event["arn"],)

    return response["Tags"]
